src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,265 @@
+            // For http, only http_proxy and generic variables are used.
+
+            Uri httpProxy = GetUriFromString(Environment.GetEnvironmentVariable(envHttpProxyLC));
+            Uri httpsProxy = GetUriFromString(Environment.GetEnvironmentVariable(envHttpsProxyLC)) ??
+                             GetUriFromString(Environment.GetEnvironmentVariable(envHttpsProxyUC));
+
+            if (httpProxy == null || httpsProxy == null)
+            {
+                Uri allProxy = GetUriFromString(Environment.GetEnvironmentVariable(envAllProxyLC)) ??
+                                GetUriFromString(Environment.GetEnvironmentVariable(envAllProxyUC));
+
+                if (httpProxy == null)
+                {
+                    httpProxy = allProxy;
+                }
+                if (httpsProxy == null)
+                {
+                    httpsProxy = allProxy;
+                }
+            }
+
+            // Do not instantiate if nothing is set.
+            // Caller may pick some other proxy type.
+            if (httpProxy == null && httpsProxy == null)
+            {
+                return null;
+            }
+
+            return new HttpEnvironmentProxy(httpProxy, httpsProxy, Environment.GetEnvironmentVariable(envNoProxyLC));
+        }
+
+        private HttpEnvironmentProxy(Uri httpProxy, Uri httpsProxy, string bypassList)
+        {
+            NetworkCredential httpCred = null;
+            NetworkCredential httpsCred = null;
+            bool sameAuth = false;
+
+            if (httpProxy != null)
+            {
+                _http = httpProxy;
+                httpCred = GetCredientialsFromString(_http.UserInfo);
+            }
+            if (httpsProxy != null)
+            {
+                _https = httpsProxy;
+                httpsCred = GetCredientialsFromString(_https.UserInfo);
+                if ((_http != null) && (_http.UserInfo == _https.UserInfo))
+                {
+                    sameAuth = true;
+                }
+            }
+            if (httpCred != null || httpsCred != null)
+            {
+                // If only one protocol is set or both protocols have same auth,
+                // use standard credential object
+                if (sameAuth || httpCred == null || httpsCred == null)
+                {
+                    _credientials = httpCred ?? httpsCred;
+                }
+                else
+                {
+                    // use wrapper so we can decide later based on uri
+                    _credientials = new HttpEnvironmentProxyCredientials(httpCred, httpsCred);
+                }
+            }
+
+            if (!string.IsNullOrWhiteSpace(bypassList))
+            {
+                string[] list = bypassList.Split(',');
+                List<string> tmpList = new List<string>();
+
+                foreach (string value in list)
+                {
+                    string tmp = value.Trim();
```

Comment:
Can there be whitespace within the string itself? like ```string with whitespace```

Reply:
no, I don't think so. valid DNS name (or IP) has letters, numbers, dots and dashes. (and underscore even if that violates original RFC)

Reply:
You can just parse the bypassList string yourself in O(n), without having to use List and array..

Reply:
```c#
        StringBuilder sb = new StringBuilderCache.Acquire(MAX_DNS_NAME_LENGTH);
        int i = 0;
        while (i < bypassList.Length)
        {
            // Skip whitespace.
            while (i < bypassList.Length && char.IsWhiteSpace(bypassList[i]))
                i++;

            // Add char chars till ',' or whitespace is found.
            while (i < bypassList.Length && bypassList[i] != ',' && !char.IsWhiteSpace(bypassList[i]))
            {
                sb.Append(bypassList[i]);
                i++;
            }

            // Skip trailing whitespace or till end of string.
            while (i < bypassList.Length && bypassList[i] != ',')
                i++;

            // Skip the ,
            i++;
            // Clear the stringbuilder
            string value = sb.ToString();
            if (value.Length != 0)
                _bypass.Add(value);

            sb.Clear();
        }

        StringBuilderCache.Release(sb);

```